src/mono/mono/mini/interp/interp.c

```diff
@@ -5218,9 +5235,11 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 		MINT_IN_CASE(MINT_SAFEPOINT)
 			/* Do synchronous checking of abort requests */
 			EXCEPTION_CHECKPOINT;
+			context_set_safepoint_frame (context, frame);
```

Comment:
MINT_SAFEPOINT is one the most common opcodes, we should try not to increase the work done on the fastpath. Could we check for mono_polling_required here ?